core/src/test/java/greencity/controller/ShoppingListItemControllerTest.java

```diff
@@ -0,0 +1,174 @@
+                        .param("habitId", habitId.toString())
+                        .param("shoppingListItemId", shoppingListItemId.toString())
+                        .principal(principal))
+                .andExpect(status().isOk());
+
+        verify(shoppingListItemService).deleteUserShoppingListItemByItemIdAndUserIdAndHabitId(
+                shoppingListItemId, userVO.getId(), habitId);
+    }
+
+    @Test
+    public void updateUserShoppingListItemStatusTest() throws Exception {
+        Long userShoppingListItemId = 1L;
+        locale = Locale.US;
+
+        mockMvc.perform(patch(shoppingListItemControllerLink+"/{userShoppingListItemId}", userShoppingListItemId)
+                        .principal(principal))
+                .andExpect(status().isCreated());
+
+        verify(shoppingListItemService).updateUserShopingListItemStatus(userVO.getId(), userShoppingListItemId,locale.getLanguage());
+    }
+
+    @Test
+    public void updateUserShoppingListItemStatusWithStatusTest() throws Exception {
+        Long userShoppingListItemId = 1L;
+        String status = "DONE";
+        locale = Locale.US;
+
+        mockMvc.perform(patch(shoppingListItemControllerLink+"/{userShoppingListItemId}/status/{status}",
+                        userShoppingListItemId, status)
+                        .principal(principal))
+                .andExpect(status().isOk());
+
+        verify(shoppingListItemService).updateUserShoppingListItemStatus(userVO.getId(), userShoppingListItemId, locale.getLanguage(), status);
+    }
+
+    @Test
+    public void findInProgressByUserIdTest() throws Exception {
+        Long userId = 1L;
+        String languageCode = "en";
+        List<ShoppingListItemDto> inProgressItems = Collections.singletonList(new ShoppingListItemDto());
+
+        when(shoppingListItemService.findInProgressByUserIdAndLanguageCode(userId, languageCode))
+                .thenReturn(inProgressItems);
+
+        mockMvc.perform(get(shoppingListItemControllerLink + "/{userId}/get-all-inprogress", userId)
+                        .param("lang", languageCode)
+                        .principal(principal))
+                .andExpect(status().isOk());
+
+        verify(shoppingListItemService).findInProgressByUserIdAndLanguageCode(userId, languageCode);
+    }
+
+    @Test
+    public void bulkDeleteUserShoppingListItemsTest1() throws Exception {
+        String commaSeparatedIds = "1,2,3";
+        List<Long> deletedIds = Arrays.asList(1L, 2L, 3L);
+
+        when(shoppingListItemService.deleteUserShoppingListItems(commaSeparatedIds)).thenReturn(deletedIds);
+
+        mockMvc.perform(delete(shoppingListItemControllerLink + "/user-shopping-list-items")
+                        .param("ids", commaSeparatedIds)
+                        .principal(principal))
+                .andExpect(status().isOk());
+
+        verify(shoppingListItemService).deleteUserShoppingListItems(commaSeparatedIds); // Verify that the service method was called
```

Comment:
I think it's better to avoid comments in the code.

Reply:
Thanks for the advice. Comment deleted.